packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -550,3 +555,75 @@ export function useLegacyFlowWithoutGroupsForSnapshotFetch(
 ): boolean {
 	return loadingGroupIds === undefined;
 }
+
+// back-compat: GitHub #9653
+const isFluidPackage = (pkg: Record<string, unknown>): boolean =>
+	typeof pkg === "object" && typeof pkg?.name === "string" && typeof pkg?.fluid === "object";
+
+/**
+ * Appends the store locator properties to the provided base URL. This function is useful for scenarios where an application
+ * has a base URL (for example a sharing link) of the Fluid file, but does not have the locator information that would be used by Fluid
+ * to load the file later.
+ * @param baseUrl - The input URL on which the locator params will be appended.
+ * @param resolvedUrl - odsp-driver's resolvedURL object.
+ * @param dataStorePath - The relative data store path URL.
+ * For requesting a driver URL, this value should always be '/'. If an empty string is passed, then dataStorePath
+ * will be extracted from the resolved url if present.
+ * @param packageInfoSource - Information of the package connected to the URL
+ * @param context - Context for given resolved URl
+ * @param appName - Name of the application that of the URL
+ * @returns The provided base URL appended with odsp-specific locator information
+ */
+export function appendNavParam(
+	baseUrl: string,
+	resolvedUrl: IResolvedUrl,
+	dataStorePath: string,
+	containerPackageName?: string,
+	context?: string,
+	appName?: string,
```

Comment:
why not just take and IOdspResolvedUrl directly? i call site gets everything off the resolved url anyway, so you should need anything but the IOdspResolvedUrl

Reply:
Do you mean that the function should just receive the base url and the IOdspResolvedUrl and extract all the info within the function?

Reply:
So, you have containerPackageName, context and appName already on the resolved URL. So, you can get it from there instead of separate arguments.